src/mongrove/core.clj

```diff
@@ -178,6 +181,8 @@
   (let [opts (merge default-opts opts)
         {:keys [read-preference read-concern write-concern
                 retry-reads retry-writes]} opts
+        credential (when (some? (and username password))
```

Comment:
I think the `some?` is redundant. Can you verify that we really need this?

Reply:
Please squash this commit.

Reply:
Thank you for the review. This is done.